### PR TITLE
Async Migrations: If sql error include sql ran in the message + handle error col limit

### PR DESCRIPTION
**Pull request by tiina303**
## Changes

Example error in the UI from local testing:
Screen Shot 2022-01-26 at 00 38 58

1. handle error column length limits in the DB. Note that in the logs & email we still see the full error, before we'd fail with 
```
...
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(400)
...
```
2. Change the sql added in https://github.com/PostHog/posthog/pull/8080 and points to a problem that sometimes we could get cryptic messages if we don't know the exact sql, so changed it so we include it in the error message instead.

## How did you test this code?

CI + manual testing:
1. modified a migration to have invalid SQL and looked at the error message I got in the UI (see pic above)

### CODE REVIEW

**Comment by yakkomajuri** [comment]
What's the problem with just holding that SQL in the class though? It's not just for errors, it _really_ helps when testing debugging dynamically generated ops particularly

**Comment by tiina303** [comment]
> What's the problem with just holding that SQL in the class though? It's not just for errors, it really helps when testing debugging dynamically generated ops particularly 

Ok there are two separate things here:
1. Do we agree that adding sql to the error message would be helpful?
2. Should we remove sql from the `AsyncMigrationOperation` class?

For (2) Could you elaborate on how you tested it and why the `sql` param there was helpful, as the only place for me that I saw it being useful is the error message.
I don't like it in that class as it adds confusion and would likely be misused, e.g. someone would do `AsyncMigrationOperartion(sql=...)` which wouldn't work.

File: posthog/async_migrations/utils.py [hunk 1]
```diff
@@ -25,14 +25,20 @@ def execute_op(op: AsyncMigrationOperation, uuid: str, rollback: bool = False):
 def execute_op_clickhouse(sql: str, query_id: str, timeout_seconds: int):
     from ee.clickhouse.client import sync_execute
 
-    sync_execute(f"/* {query_id} */ " + sql, settings={"max_execution_time": timeout_seconds})
+    try:
+        sync_execute(f"/* {query_id} */ " + sql, settings={"max_execution_time": timeout_seconds})
+    except Exception as e:
+        raise Exception(f"Failed to execute ClickHouse op: sql={sql}, query_id={query_id}, exception={str(e)}")
```
Comment (yakkomajuri) [inline posthog/async_migrations/utils.py:31]: won't this get messy? maybe we want some new lines?

File: posthog/async_migrations/definition.py [hunk 2]
```diff
@@ -40,9 +40,6 @@ def __init__(
         # Defaults to a no-op ("") - None causes a failure to rollback
         self.rollback_fn = rollback_fn
 
-        # If the operation specifies raw SQL, add this as a property for easier debugging
-        self.sql = sql
```
Comment (yakkomajuri) [inline posthog/async_migrations/definition.py:44]: I'm still strongly opposed to this. We have 2 migrations generating SQL dynamically and we don't have that SQL easily available anywhere. Even with the `current_operation_index` it's very hard to determine what the SQL for the operation is without this or something similar. 

We can rename to `_sql` or even `__sql` if it makes you more comfortable

**Review by yakkomajuri — CHANGES_REQUESTED** [review]

File: posthog/async_migrations/utils.py (hunk 1 above)
Comment (tiina303) [reply posthog/async_migrations/utils.py:31]: yes, new lines are nice - we need them in the UI too later

File: posthog/async_migrations/utils.py [hunk 3]
```diff
@@ -158,7 +164,7 @@ def execute_update():
         else:
             instance.refresh_from_db()
         if error is not None:
-            AsyncMigrationError.objects.create(async_migration=instance, description=error).save()
+            AsyncMigrationError.objects.create(async_migration=instance, description=error[:398]).save()
```
Comment (yakkomajuri) [inline posthog/async_migrations/utils.py:167]: I guess this will go away?

**Review by yakkomajuri — APPROVED** [review]

File: posthog/async_migrations/utils.py (hunk 3 above)
Comment (tiina303) [reply posthog/async_migrations/utils.py:167]: yup, they are stacked though

**Pull request merged**